xbmc/ServiceManager.cpp

```diff
@@ -151,6 +152,9 @@ bool CServiceManager::InitStageTwo(const CAppParamParser &params)
 
   m_weatherManager.reset(new CWeatherManager());
 
+  // register settings callbacks
+  GetSettings().GetSettingsManager()->RegisterCallback(m_weatherManager.get(), { CSettings::SETTING_WEATHER_ADDON, CSettings::SETTING_WEATHER_ADDONSETTINGS });
```

Comment:
This should be done inside weatherManager (maybe ctor/dtor). ServiceManager should not bother with internals.

Reply:
done